beacon_node/network/src/sync/block_lookups/single_block_lookup.rs

```diff
@@ -172,17 +172,11 @@ impl<T: BeaconChainTypes> SingleBlockLookup<T> {
         )
     }
 
-    /// Add all given peers to both block and blob request states.
-    pub fn add_peer(&mut self, peer_id: PeerId) {
-        self.block_request_state.state.add_peer(&peer_id);
-        self.blob_request_state.state.add_peer(&peer_id);
-    }
-
-    /// Add all given peers to both block and blob request states.
-    pub fn add_peers(&mut self, peers: &[PeerId]) {
-        for peer in peers {
-            self.add_peer(*peer);
-        }
+    /// Add peer to all request states. The peer must be able to serve this request.
+    /// Returns true if the peer was newly inserted into some request state.
+    pub fn add_peer(&mut self, peer_id: PeerId) -> bool {
+        self.block_request_state.state.add_peer(&peer_id)
+            || self.blob_request_state.state.add_peer(&peer_id)
```

Comment:
Is it intentional to short circuit this? i.e. if `self.block_request_state.state.add_peer(&peer_id)` returns true then it won't add peer to `blob_request_state`. 
If it is, could we add some reasoning here as docs?

Reply:
Nice pick! Definitely not intended behaviour